packages/aserto/src/aserto/client/directory/__init__.py

```diff
@@ -0,0 +1,180 @@
+                    subject=ObjectIdentifier(type=subject_type, key=subject_key),
+                    relation=RelationTypeIdentifier(name=relation_type, object_type=object_type),
+                ),
+                page=PaginationRequest(page),
+            )
+        )
+        return response
+
+    def get_relation(
+        self,
+        subject_type: str,
+        subject_key: str,
+        object_type: str,
+        object_key: str,
+        relation_type: str,
+    ) -> Relation:
+        response = self.reader.GetRelation(
+            GetRelationRequest(
+                param=RelationIdentifier(
+                    object=ObjectIdentifier(type=object_type, key=object_key),
+                    subject=ObjectIdentifier(type=subject_type, key=subject_key),
+                    relation=RelationTypeIdentifier(name=relation_type, object_type=object_type),
+                )
+            )
+        )
+        return response["results"]
+
+    def set_relation(self, relation: Relation) -> Relation:
+        response = self.writer.SetRelation(SetRelationRequest(relation=relation))
+        return response["result"]
+
+    def delete_relation(
+        self,
+        subject_type: str,
+        subject_key: str,
+        object_type: str,
+        object_key: str,
+        relation_type: str,
+    ) -> None:
+        relation_identifier = RelationIdentifier(
+            object=ObjectIdentifier(type=object_type, key=object_key),
+            subject=ObjectIdentifier(type=subject_type, key=subject_key),
+            relation=RelationTypeIdentifier(name=relation_type, object_type=object_type),
+        )
+        self.writer.DeleteRelation(DeleteRelationRequest(param=relation_identifier))
+
+    def check_relation(
+        self,
+        subject_type: str,
+        subject_key: str,
+        object_type: str,
+        object_key: str,
+        relation_type: str,
+    ) -> bool:
+        response = self.reader.CheckRelation(
+            CheckRelationRequest(
+                object=ObjectIdentifier(type=object_type, key=object_key),
+                subject=ObjectIdentifier(type=subject_type, key=subject_key),
+                relation=RelationTypeIdentifier(name=relation_type, object_type=object_type),
+            )
+        )
+        return response["check"]
+
+    def check_permission(
+        self,
+        subject_type: str,
+        subject_key: str,
+        object_type: str,
+        object_key: str,
+        permission: str,
+    ) -> bool:
+        response = self.reader.CheckPermission(
+            CheckPermissionRequest(
+                object=ObjectIdentifier(type=object_type, key=object_key),
+                subject=ObjectIdentifier(type=subject_type, key=subject_key),
+                permission=PermissionIdentifier(name=permission),
+            )
+        )
+        return response["check"]
+
+    def close_channel(self) -> None:
+        self._channel.close()
+
+    def _metadata(self) -> Tuple:
```

Comment:
@ananyakaushik should this be called when doing the request? maybe that's why you get authentication errors

Reply:
I don't use this method anywhere in the todo app and commenting it out still results in the same authentication error

```
raise _InactiveRpcError(state)
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.FAILED_PRECONDITION
	details = "E40001 authentication failed"
	debug_error_string = "UNKNOWN:Error received from peer ipv4:20.242.152.122:8443 {grpc_message:"E40001 authentication failed", grpc_status:9, created_time:"2023-06-22T10:07:36.880161-07:00"}"
```

Reply:
this needs to be sent when you create the grpc client, https://github.com/aserto-dev/aserto-python/blob/508661f5ad9bab42b3c16cf7f3f9e1843c8981b2/packages/aserto/src/aserto/api/authorizer/grpc.py#L59